go/pkg/cas/upload.go

```diff
@@ -82,16 +89,31 @@ func (c *Client) Upload(ctx context.Context, inputC <-chan *UploadInput) (stats
 	// We might run into the request size limits only if we have >100K digests.
 	u.checkBundler.BundleCountThreshold = u.FindMissingBlobsBatchSize
 
+	// Initialize batchBundler, which uploads blobs in batches.
+	u.batchBundler = bundler.NewBundler(&repb.BatchUpdateBlobsRequest_Request{}, func(subReq interface{}) {
+		// Handle errors and context cancelation via errgroup.
+		eg.Go(func() error {
+			return u.uploadBatch(ctx, subReq.([]*repb.BatchUpdateBlobsRequest_Request))
+		})
+	})
+	// Limit the sum of sub-request sizes to (maxRequestSize - requestOverhead).
+	// 4MiB is the default gRPC request size limit.
+	// TODO: retrieve it using u.conn.GetMethodConfig instead. Currently it would break tests.
+	// Subtract 1KB to be on the safe side.
+	u.batchBundler.BundleByteLimit = 4*1024*1024 - int(marshalledFieldSize(int64(len(c.InstanceName)))) - 1000
```

Comment:
Updated the TODO to retrieve from capabilities instead. Is this what you meant?

Reply:
Thanks. Just to make sure we're on the same page, we should load from Capabilities if the `startup_capabilities` flag is set: https://github.com/bazelbuild/remote-apis-sdks/blob/2dacc28301fb5eb8af27adf40456cac474f6f0e6/go/pkg/flags/flags.go#L60

Reply:
It looks like that package defines a global flag. I don't think that would be OK for many applications that might want to use this package. For example, this would be surprising to see in our servers - seems invasive. In our CLI, we use a specific subcommand package, and we very rarely use global flags.

Would it be OK to have a boolean field in `ClientConfig` instead? The code already has a TODO on capabilities, so I it would be done as a part of that TODO

Reply:
I've updated that TODO

Reply:
Sure the `cas` package should not rely on a global flag, the current client package doesn't either. Adding a booolean and a TODO sgtm. Thanks!